### PR TITLE
DBTP-343 Correct Handling of hyphens when getting Load Balancer details

### PR DESCRIPTION
**Pull request by WillGibson**
For https://uktrade.atlassian.net/browse/DBTP-343

I should have tested my code manually, there were extra hyphen separated things on the end of the cluster and service ARNs in the real world which are not included in the unit tests.

### CODE REVIEW

File: commands/dns_cli.py [hunk 1]
```diff
@@ -324,9 +324,15 @@ def check_r53(domain_session, project_session, domain, base_domain):
...
-    def separate_hyphenated_application_environment_and_service(hyphenated_string):
-        # The application name may be hyphenated, so we start splitting at the second from last occurrence of a hyphen
-        return hyphenated_string.rsplit("-", 2)
+    def separate_hyphenated_application_environment_and_service(
+        hyphenated_string, number_of_items_of_interest, number_of_trailing_items
+    ):
+        # The application name may be hyphenated, so we start splitting
+        # at the hyphen after the first item of interest and return the
+        # items of interest only...
+        rsplit = hyphenated_string.rsplit("-", number_of_trailing_items + number_of_items_of_interest - 1)
```
Comment (yusufsheiqh) [inline commands/dns_cli.py:333]: Can we rename this variable to something else?

File: tests/test_dns_cli.py [hunk 2]
```diff
@@ -183,7 +183,7 @@ def test_get_load_balancer_domain_and_configuration_no_clusters(capfd):
 @mock_ecs
 def test_get_load_balancer_domain_and_configuration_no_services(capfd):
     boto3.Session().client("ecs").create_cluster(
-        clusterName=f"{HYPHENATED_APPLICATION_NAME}-{ALPHANUMERIC_ENVIRONMENT_NAME}-{ALPHANUMERIC_SERVICE_NAME}"
+        clusterName=f"{HYPHENATED_APPLICATION_NAME}-{ALPHANUMERIC_ENVIRONMENT_NAME}-Cluster-c0PIlotiD3ntIF3r"
```
Comment (yusufsheiqh) [inline tests/test_dns_cli.py:186]: Shall we create a constant for `Cluster-c0PIlotiD3ntIF3r` as well? I see it is used in another test as well.

File: commands/dns_cli.py (hunk 1 above)
Comment (gabelton) [inline commands/dns_cli.py:333]: Not a blocker, but these variable names are a bit of a mouthful

**Review by gabelton — APPROVED** [review]

File: commands/dns_cli.py (hunk 1 above)
Comment (WillGibson) [reply commands/dns_cli.py:333]: These were only meant to be temporary for breakpoints while I figured it out. Gone now.

Comment (WillGibson) [reply commands/dns_cli.py]: But you don't have to think about what they mean ;-)

I'm open to suggestions for conveying the same meaning in less words.

File: tests/test_dns_cli.py (hunk 2 above)
Comment (WillGibson) [reply tests/test_dns_cli.py]: I extracted a couple more while I was at it. Less for duplication reasons, that readability.

**Review by yusufsheiqh — APPROVED** [review]

**Pull request merged**